sqlrunner/main.py

```diff
@@ -35,7 +37,13 @@ def main(args):
 
 
 def generate_column_headers(dummy_data_file, sql_query):
-    raise NotImplementedError
+    tsql_parser = TSQL()
+    parsed = [q for q in tsql_parser.parse(sql_query) if q]
+    # expand out "*" to full column names
+    columns = qualify_columns(parsed[0], schema=None, infer_schema=True).named_selects
+    with open(dummy_data_file, "w") as f:
```

Comment:
Although I think we should return the column headers in a single-element iterable, if we were going to write them to a file, [then remember `newline=""`](https://docs.python.org/3.10/library/csv.html#csv.writer).

Reply:
I agree, having read your top-level comment it seems sensible to pass this to `write_results` based on the (non)presence of dsn and dummy data file